packages/core-sdk/resolvers/token-resolvers/resolvers/yearn/src/query/functions/getTokenComponents.ts

```diff
@@ -0,0 +1,83 @@
+import { BigInt } from "@web3api/wasm-as";
+import { Big } from "as-big";
+
+import {
+  env,
+  Ethereum_Query,
+  Input_getTokenComponents,
+  Interface_TokenComponent,
+  QueryEnv,
+  Token_Query,
+  Token_TokenType,
+} from "../w3";
+
+export function getTokenComponents(input: Input_getTokenComponents): Interface_TokenComponent {
+  if (env == null) throw new Error("env is not set");
+  const connection = (env as QueryEnv).connection;
+
+  const token = Token_Query.getToken({
+    address: input.tokenAddress,
+    m_type: Token_TokenType.ERC20,
+  }).unwrap();
+
+  if (!token) {
+    throw new Error(`Token ${input.tokenAddress} is not a valid ERC20 token`);
+  }
```

Comment:
Question: Should we throw an error inside `getToken` instead of silently returning null. This will allow consumers of getToken to handle errors the way he wants. 

We didn't have any error handling primitives before so, I wasn't throwing errors when we get an invalid response from the contract (Ex: empty string, zero address, etc) but since now we have the error handlers, I think we should let consumers handle it.